ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,316 @@
+
+import textwrap
+
+from tempfile import NamedTemporaryFile
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+
+
+class TestAutomember(IntegrationTest):
+    """
+    Tests of Automember feature.
+    Covers unmembering feature described in
+    https://pagure.io/389-ds-base/issue/50077
+    """
+    topology = 'line'
+
+    def test_modify_user_entry_unmembership(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        tasks.kinit_admin(self.master)
+        master = self.master
+        search_cmd = [
+            'ldapsearch', '-x',
+            '-D', 'cn=directory manager',
+            '-w', master.config.dirman_password,
+            '-b', 'cn=Auto Membership Plugin,cn=plugins,cn=config']
+        result = self.master.run_command(
+            search_cmd,
+            raiseonerr=False
+        )
+        assert result.returncode == 0
+        assert "automemberprocessmodifyops: on" in result.stdout_text
+
+        '#testcase setup'
+        cmdlist = [['ipa', 'group-add', '--desc="Dallas"', 'DallasUsers'],
+                   ['ipa', 'automember-add', '--type=group', 'DallasUsers'],
+                   ['ipa', 'automember-add-condition',
+                    '--key=destinationindicator', '--type=group',
+                    '--inclusive-regex=Dallas', 'DallasUsers'],
+                   ['ipa', 'user-add', '--first=first', '--last=user',
+                    'user1', '--setattr=destinationindicator=Dallas']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
+
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'DallasUsers'],
+        )
+        assert "user1" in result.stdout_text
+
+        '#Modifying the user group so that it becomes part of new group'
+        cmdlist = [['ipa', 'group-add', '--desc="Houston"', 'HoustonUsers'],
+                   ['ipa', 'automember-add', '--type=group', 'HoustonUsers'],
+                   ['ipa', 'automember-add-condition',
+                    '--key=destinationindicator', '--type=group',
+                    '--inclusive-regex=Houston', 'HoustonUsers'],
+                   ['ipa', 'user-mod', 'user1',
+                    '--setattr=destinationindicator=Houston']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
+
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'HoustonUsers'],
+        )
+        assert "user1" in result.stdout_text
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'DallasUsers'],
+        )
+        assert "user1" not in result.stdout_text
+
+        '#Deleting the user entry and checking its not part of group anymore'
+        self.master.run_command(['ipa', 'user-del', 'user1'])
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'HoustonUsers'],
+        )
+        assert "user1" not in result.stdout_text
+
+        '#testcase cleanup'
```

Comment:
The single quotes can be removed.

Reply:
To make the test more robust, it can be written like this:
try:
   [do the calls]
finally:
   [do the cleanup]

This way, even if there are some failures, the cleanup will be executed.